### PR TITLE
chore: Bump deps 2022.10.31

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Bump deps 2022.10.31

### CODE REVIEW

File: src/event-timer.ts [hunk 1]
```diff
@@ -161,20 +179,21 @@ export class EventTimer {
 				},
 			],
 		};
+		const localWindow = window as unknown as LocalWindow;
```
Comment (arelra) [inline src/event-timer.ts:182]: Bumping TypeScript from v4.7.4 to v4.8.4 means we lose the `NavigatorNetworkInformation` type as it is not supported by the majority of browsers.

Not so happy about this fix. Overriding the window.navigator type does not seem possible as you then run into:

`Subsequent property declarations must have the same type` errors for `lib.dom.d.ts`

Is there a better solution?

Comment (chrislomaxjones) [reply src/event-timer.ts:182]: Could you try overwriting the `Navigator` type in the global scope directly? Something like

```typescript
declare global {
	interface Navigator {
		readonly connection: NetworkInformation;
	}
}
```

File: src/event-timer.ts [hunk 2]
```diff
@@ -33,6 +33,24 @@ interface PageEventStatus {
 	commercialModulesLoaded: boolean;
 }
 
+type ConnectionType = "bluetooth" | "cellular" | "ethernet" | "mixed" | "none" | "other" | "unknown" | "wifi";
```
Comment (chrislomaxjones) [inline src/event-timer.ts:36]: I thought double quotes would get flagged by Prettier?

File: src/event-timer.ts (hunk 1 above)
Comment (arelra) [reply src/event-timer.ts:182]: Nice idea thanks! I've tried it out but that still doesn't override `Navigator` from `lib.dom.d.ts` and you get a similar error message:

`All declarations of 'navigator' must have identical modifiers`

I shall pick your TS brain IRL :)

File: src/event-timer.ts (hunk 2 above)
Comment (arelra) [reply src/event-timer.ts:36]: That is surprising. It isn't flagging and CI passes.

File: src/event-timer.ts (hunk 1 above)
Comment (chrislomaxjones) [reply src/event-timer.ts:182]: I've got a [branch here](https://github.com/guardian/commercial-core/compare/test-navigator-types) with a minimal reproduction, but am not too sure it's exactly what we're after, so happy to discuss!

File: src/event-timer.ts (hunk 2 above)
Comment (chrislomaxjones) [reply src/event-timer.ts:36]: Strange, I now wonder if we ever actually check Prettier in CI, or just rely on people auto-formatting on save 🤔

**Review by chrislomaxjones — APPROVED** [review]
Always keeping on top of dependencies @arelra 👍

**Pull request merged**